docs/.sphinx/constraints.txt

```diff
@@ -0,0 +1,3 @@
+# watchfiles, a direct dependency of sphinx-autobuild, introduces dependency on
+# rustc 1.77 in version 1.1.0.
+watchfiles<1.1.0
```

Comment:
Overall I agree that we need a mechanism to lock down dependencies, so that we don't end up with broken builds down the line. However in this specific case, it seems that all we need is a `rustc <= 1.77` and `noble` provides rust toolchain up to `1.85` (for all our relevant archs). Or am I missing something?